assignments/introductory-assignments.md

```diff
@@ -488,14 +582,767 @@ A digital circuit computer with 2GB of memory can distinguish 2^{17 179 869 184}
 - How many different states can a digital computer with 4GB of memory have?
 - How many different states can a digital computer with 80GB of storage have?
 
+## Signed and unsigned integer
+
+Bytes can represent numbers as well as alphabetic characters. But if we want to represent negative numbers, we have to use a sign bit.
+
+### One's complement
+
+The [one's complement](https://en.wikipedia.org/wiki/Signed_number_representations) can be used to represent negative numbers.
+
+**8-bit representation**
+
+We only use 7 bits for the value and the **MSB** for the sign with a simple sign bit.
+If the **MSB** is a 1, it represents a `-` else it is a `+`.
+
+
+| bit values |  +/-  |  64   |  32   |  16   |   8   |   4   |   2   |   1   | Decimal Value |
+| :--------- | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | ------------: |
+| bits       |   0   |   0   |   1   |   0   |   1   |   0   |   0   |   1   |            41 |
+| bits       |   1   |   0   |   1   |   0   |   1   |   0   |   0   |   1   |           -41 |
+
+
+**Reason why we don't use the one's complement in practice:**
+
+| bit values |  +/-  |  64   |  32   |  16   |   8   |   4   |   2   |   1   | Decimal Value |
+| :--------- | :---: | :---: | :---: | :---: | :---: | :---: | :---: | :---: | ------------: |
+| bits       |   0   |   0   |   0   |   0   |   0   |   0   |   0   |   0   |             0 |
+| bits       |   1   |   0   |   0   |   0   |   0   |   0   |   0   |   0   |            -0 |
+
+We get two representations for 0, but we only need one.
+
+### Two's complement
+
+Integers are also commonly represented by using [two's complement](https://en.wikipedia.org/wiki/Two%27s_complement) notation, in which the left-most bit indicates the sign: If the leading bit is 1, we subtract 2^{n} to get the [integer](https://en.wikipedia.org/wiki/Integer_(computer_science)) corresponding to an n-bit number in this notation. Two's complement is used to change subtraction to addition by using the "complementary" of a number.
```

Comment:
Satz ändern damit man "comple.." nicht machen muss. Harte Aussage daraus machen

Two's complement allows to perform substr using addition. For example 42-7 is equal to 42 + (-7) and minus 7 ist the two's complement of minus seven.

Reply:
right

Reply:
```suggestion
Integers are also commonly represented by using [two's complement](https://en.wikipedia.org/wiki/Two%27s_complement) notation, in which the left-most bit indicates the sign: If the leading bit is 1, we subtract 2^{n} to get the [integer](https://en.wikipedia.org/wiki/Integer_(computer_science)) corresponding to an n-bit number in this notation. Two's complement allows performing subtraction using an addition. For example, the arithmetic operation 42 - 7 is equal to 42 + (-7), and plus-minus 7 is the two's complement of minus 7.
```